cruise.umple/src/Parser_Code.ump

```diff
@@ -1785,6 +1860,20 @@ class ErrorTypeSingleton
     parse(new TextParser(filename, input));
   }
 
+  private static <T extends Closeable> void closeAsRequired(T reader){
```

Comment:
Yeah I'm not 100% sure on how to handle this one.

It was originally in the `SampleFileWriter`, and I wasn't really sure of any other way of handling this. `SampleFileWriter` is used heavily in the tests for the core compiler, and itself exists in a different namespace in Umple. Given that I was trying to remove all ties from the parser, this was one thing I had to extract.

The only reason it's repeated twice is because I can't think of a good reason to add a parent class or trait with this capability to both classes that need them since `Parser` and `ErrorTypeSingleton` don't really share much in common. 

Normally I'm opposed to duplication. That said, given that the method's functionality is relatively simple, that it's only a few lines long, and because it lacks any real ties between the two objects, I figured that this wouldn't really be that huge of a problem.